src/lib/bootloader/mbr_update.rb

```diff
@@ -181,20 +198,14 @@ def extended_partition_num(disk)
     #  containing device (eg. "/dev/hda4"), disk (eg. "/dev/hda") and
     #  partition number (eg. 4)
```

Comment:
empty one can be returned only for very strange storage setup like only swap partitions on disk or ones that cannot be activated, so empty one means "activate nothing".

I do not get your question. It return device for mbr ( disk device ) and number of its partition (giving together PBR - partition boot record ) which is needed for parted, as parted need disk and partition number.